source/lib.ts

```diff
@@ -0,0 +1,271 @@
+// /source/lib.ts
+// The option parser and rate limiting middleware
+
+import Express from 'express'
+
+import { Options, AugmentedRequest, RateLimitRequestHandler } from './types.js'
+import MemoryStore from './memory-store.js'
+
+/**
+ * Adds the defaults for options the user has not specified.
+ *
+ * @param options {Options} - The options the user specifies
+ *
+ * @returns {Options} - A complete configuration object
+ */
+const parseOptions = (passedOptions: Partial<Options>): Options => {
+	const {
+		// Default window of time is 1 minute
+		windowMs = 60 * 1000,
+		// The default storage provider used is a {@link MemoryStore}
+		store = new MemoryStore(windowMs),
+	} = passedOptions
+
+	// Now add the defaults for the other options
+	const options: Options = {
```

Comment:
Why is this two separate blocks?

Reply:
Not sure... will combine it with the second block